### PR TITLE
Fix CI on 0.9.x

### PR DESCRIPTION
**Pull request by twoeths**
+ Fix e2e
+ Spec tests should be fixed in a separate PR

### CODE REVIEW

**Review by wemeetagain — APPROVED** [review]

File: packages/eth2.0-state-transition/src/util/duties.ts [hunk 1]
```diff
@@ -25,8 +25,6 @@ export function getCommitteeAssignment(
   validatorIndex: ValidatorIndex
 ): CommitteeAssignment {
 
-  const nextEpoch = getCurrentEpoch(config, state) + 1;
-  assert(epoch <= nextEpoch);
```
Comment (mpetrunic) [inline packages/eth2.0-state-transition/src/util/duties.ts:29]: why did you remove this?

File: packages/lodestar-validator/src/services/attestation.ts [hunk 2]
```diff
@@ -106,10 +106,21 @@ export class AttestationService {
...
+      if (!isFound && equals(this.config.types.Attestation, attestation, wireAttestation)) {
+        isFound = true;
+      }
+      if (equals(this.config.types.AttestationData, wireAttestation.data, attestation.data)) {
+        compatibleAttestations.push(wireAttestation);
+      }
+    }
+    if (!isFound) {
+      compatibleAttestations.push(attestation);
+    }
```
Comment (mpetrunic) [inline packages/lodestar-validator/src/services/attestation.ts]: I think this achieves same this but it's more readable:
```javascript
const compatibleAttestations = wireAttestations.filter((wireAttestation) => {
   return equals(this.config.types.AttestationData, wireAttestation.data, attestation.data)
                 // prevent including aggregator attestation twice
                 && ! equals(this.config.types.Attestation, attestation, wireAttestation)
});
 compatibleAttestations.push(attestation);
```

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: packages/eth2.0-state-transition/src/util/duties.ts (hunk 1 above)
Comment (twoeths) [reply packages/eth2.0-state-transition/src/util/duties.ts:29]: yeah I'm concerning about this too. When I run e2e `dev.test.ts` it failed at this line.
+ Given SLOTS_PER_EPOCH=5
+ At slot 5 => epoch = 1, the validator asks for duties of next epoch, which is 2. In the above context `epoch=2`
+ But `nextEpoch` is evaluated as 1 because it only see blocks 0,1,2,3,4 => `getCurrentEpoch` evaluated to 0

I don't see anything wrong here so I have to comment this out. Do you think this is because we changed something in validator?

Comment (mpetrunic) [reply packages/eth2.0-state-transition/src/util/duties.ts:29]: Hmm, I think it's okay to bump next epoch = currentEpoch +2 as this is just to protect us from doing way too many epoch transitions if there is malicious user

**Review by wemeetagain — APPROVED** [review]
